library/core/src/num/dec2flt/mod.rs

```diff
@@ -204,155 +194,69 @@ impl ParseFloatError {
     }
 }
 
-#[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Display for ParseFloatError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         self.__description().fmt(f)
     }
 }
 
-fn pfe_empty() -> ParseFloatError {
+pub fn pfe_empty() -> ParseFloatError {
     ParseFloatError { kind: FloatErrorKind::Empty }
 }
 
-fn pfe_invalid() -> ParseFloatError {
+pub fn pfe_invalid() -> ParseFloatError {
     ParseFloatError { kind: FloatErrorKind::Invalid }
 }
 
-/// Splits a decimal string into sign and the rest, without inspecting or validating the rest.
-fn extract_sign(s: &str) -> (Sign, &str) {
-    match s.as_bytes()[0] {
-        b'+' => (Sign::Positive, &s[1..]),
-        b'-' => (Sign::Negative, &s[1..]),
-        // If the string is invalid, we never use the sign, so we don't need to validate here.
-        _ => (Sign::Positive, s),
-    }
+/// Converts a `BiasedFp` to the closest machine float type.
+/// Exposed only for testing.
```

Comment:
Is this some leftover? I'm not seeing `biased_fp_to_float` used anywhere in the tests.

Reply:
Yep, this was a leftover. This has been fixed [here](https://github.com/Alexhuszagh/rust/blob/f006d9db3153724274554907132409ca2e8dd83a/library/core/src/num/dec2flt/mod.rs#L214-L219).

```rust
/// Converts a `BiasedFp` to the closest machine float type.
fn biased_fp_to_float<T: RawFloat>(x: BiasedFp) -> T {
    let mut word = x.f;
    word |= (x.e as u64) << T::MANTISSA_EXPLICIT_BITS;
    T::from_u64_bits(word)
}
```